joomla/joomla-cms
# Whitelist/Blacklist mail domain and TLD allowed

### Summary of Changes
This PR adds a restriction in case the user registration is needed, only for a group of mail domain and/or TLD, or block for a group of mail domain and/or TLD.

### Testing Instructions for the option "Whitelist"
1. Go to the configs in the user manager
2. Select a _Whitelist_ option in the field **Email/TLD Restriction**. Fill the field **Email/TLD List** and save.
3. Go to the frontend/backend, and try to register/edit a user with a different domain/TLD than the field **Email/TLD List**. Also, check this is valid for the **Edit Account** option in the top right corner of the admin area. 
4. Go to the frontend/backend, and try to register/edit a user with a domain/TLD which is on the list **Email/TLD List**. Also, check this is valid for the "Edit Account" option in the top right corner of the admin area. 

### Expected result
An error message will show up, saying, "Your mail domain xxx is not allowed", refusing registration in case you try to register with a different domain/TLD. Or the new registration is accepted otherwise.

### Testing Instructions for the option "Blacklist"
1. Go to the configs in the user manager
2. Select a _Blacklist_ option in the field **Email/TLD Restriction**. Fill the field **Email/TLD List** and save.
3. Go to the frontend/backend, and try to register/edit a user with a different domain/TLD than the field **Email/TLD List** . Also, check this is valid for the **Edit Account** option in the top right corner of the admin area. 
3. Go to the frontend/backend, and try to register/edit a user with a domain/TLD which is on the list **Email/TLD List**. Also, check this is valid for the **Edit Account** option in the top right corner of the admin area. 

### Expected result
An error message will show up, saying, "Your mail domain xxx is not allowed", refusing registration in case you try to register with a domain/TLD, which it is within of the field **Email/TLD List**. Or the new registration is accepted otherwise.

### Documentation Changes Required
New com_users config added.
